qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -445,6 +445,36 @@ def acquire(self, mode=None, samples_per_record=None,
         samples_per_record = self.samples_per_record.raw_value
         records_per_buffer = self.records_per_buffer.raw_value
 
+        # bits per sample
+        _, bits_per_sample = self.api.get_channel_info_(self._handle)
+        # Whole bytes per sample
+        bytes_per_sample = (bits_per_sample + 7) // 8
+        # bytes per record
+        bytes_per_record = bytes_per_sample * samples_per_record
+
+        # channels
+        channels_binrep = self.channel_selection.raw_value
+        number_of_channels = self.get_num_channels(channels_binrep)
+
+        # bytes per buffer
+        bytes_per_buffer = (bytes_per_record *
+                            records_per_buffer * number_of_channels)
+
+        sample_type = ctypes.c_uint16 if bytes_per_sample > 1 else ctypes.c_uint8
+
+        max_buffer_size = 64*1024*1024
+        # 64 MB see docs of AlazarBeforeAsyncRead
+        # http://www.alazartech.com/Support/Download%20Files/ATS-SDK-Guide-7.2.3.pdf#section*.110
+
+        requested_buffer_size = (bits_per_sample * samples_per_record *
```

Comment:
why is `bits_per_sample` with `//8` used instead of `bytes_per_sample` from above?

Reply:
Because there are 2 different considerations. The absolute size of the buffer (in this case there are 1.5 bytes per sample) 
and the padded to whole bytes per sample size. As far as I can see from the docs the internal buffer utilization needs to be calculated from the fractional value but we transfer whole bytes. I will clarify the variable names.

Reply:
The calculation with unites is 12 bits/sample / 8 bits/byte * x samples/record *y  records/ buffer which should give (12/8) * x * y bytes/buffer

Reply:
ok, thanks! So i was confusing on-board-memory buffers with host-memory buffers.

So the limit is for the on-board-memory buffer, right? and for some reason the limit does not involve number of channels, right?

And thanks for detailed comments around this formula in the newer commits!